server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java

```diff
@@ -3728,31 +3727,30 @@ public CustomTranslogDeletionPolicy(IndexSettings indexSettings, Supplier<Retent
         TranslogDeletionPolicyFactory translogDeletionPolicyFactory = CustomTranslogDeletionPolicy::new;
 
         EngineConfig config = engine.config();
-        EngineConfig configWithCustomTranslogDeletionPolicyFactory = new EngineConfig(
-            config.getShardId(),
-            config.getThreadPool(),
-            config.getIndexSettings(),
-            config.getWarmer(),
-            config.getStore(),
-            config.getMergePolicy(),
-            config.getAnalyzer(),
-            config.getSimilarity(),
-            new CodecService(null, logger),
-            config.getEventListener(),
-            config.getQueryCache(),
-            config.getQueryCachingPolicy(),
-            config.getTranslogConfig(),
-            translogDeletionPolicyFactory,
-            config.getFlushMergesAfter(),
-            config.getExternalRefreshListener(),
-            config.getInternalRefreshListener(),
-            config.getIndexSort(),
-            config.getCircuitBreakerService(),
-            config.getGlobalCheckpointSupplier(),
-            config.retentionLeasesSupplier(),
-            config.getPrimaryTermSupplier(),
-            config.getTombstoneDocSupplier()
-        );
+        EngineConfig configWithCustomTranslogDeletionPolicyFactory = new EngineConfig.Builder().setShardId(config.getShardId())
```

Comment:
There are multiple cases of taking an existing config and making a small change to it in these tests. What do you think about adding a `toBuilder()` method that would make this more concise and more clear. This case would look like:

```
config.toBuilder().setTranslogDeletionPolicyFactory(translogDeletionPolicyFactory).create();
```

Reply:
Agree, but changing all these tests to use `toBuilder` would take some time. I would like to do it as a follow-up. 

I am also thinking of using [Lombok](https://projectlombok.org/) in OpenSearch which will get rid of such boilerplate code (in other classes as well). I will create a tracking issue for the same.